presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaMetadata.java

```diff
@@ -225,6 +227,11 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
 
         log.info("startTs: %s, endTs: %s", startTs, endTs);
+        if (config.isHardLimitOn() && startTs == null && endTs == null) {
```

Comment:
1. If the user doesn't know how to specify a range, Presto should either throw an exception **directly**, or trim to the limited time range with an **explicit logging/print-out notifying** users. Siliently unexpected behavior is **bad**.

2. A design should be resilient to users' mistakes/errors, rather than to rely on users always providing safe input. Also, this is what **limitation** means: it should not happen.

If users want to query exceeding the limitation, they need to notify all stakeholders, for instance, messaing team, since it could potentially break the whole kafka service, leading to a SEV0 or SEV1.